java/src/jmri/jmrix/loconet/LocoNetThrottle.java

```diff
@@ -335,17 +335,17 @@ public void setSpeedSetting(float speed, boolean allowDuplicates, boolean allowD
             log.debug("setSpeedSetting: float speed: {} LocoNet speed: {}", speed, new_spd);
             msg.setElement(2, new_spd);
             network.sendLocoNetMessage(msg);
+
+            // reset timeout - but only if something sent on net
+            if (mRefreshTimer != null) {
+                mRefreshTimer.stop();
+                mRefreshTimer.setRepeats(true);     // refresh until stopped by dispose
+                mRefreshTimer.start();
+                log.debug("Initially starting refresh timer for slot {} address {}", slot.getSlot(), slot.locoAddr());
+            }
```

Comment:
Code looks good. But this change is a significant behavioral change, and needs a test case.

That test case needs to _fail_ under the un-modified code, but pass on the code when modified as you propose.

Reply:
Also, a modification to the "release notes" needs to be provided, so that JMRI users are made aware of the change to the _behavior_ (not the technical details!).